File: Haskell/59A.hs
import Data.Char (toUpper, toLower, isUpper)

main :: IO ()
main = getLine >>= putStrLn . solve  

solve :: String -> String
solve s = case compare u l of
  EQ -> map toLower s
  LT -> map toLower s
  GT -> map toUpper s
  where
    (u, l) = foldl (\ (u, l) s -> if isUpper s then (u + 1 , l) else (u, l + 1) ) (0, 0) s

File: CPP/59A.cpp
#include <iostream>
#include <algorithm>
#include <string>

using namespace std;

int main() {
  string str;
  cin >> str;

  size_t upcase = 0, lowcase = 0;

  for (const auto& e: str) {
    if (isupper(e)) upcase++;
    else lowcase++;
  }

  if (upcase == lowcase || upcase < lowcase)
    transform(str.begin(), str.end(), str.begin(), ::tolower);
  else
    transform(str.begin(), str.end(), str.begin(), ::toupper);

  cout << str << endl;
}